### PR TITLE
Add package.json scripty options support for npm7

### PR DESCRIPTION
**Pull request by Chocobozzz**
Closes https://github.com/testdouble/scripty/issues/93

### CODE REVIEW

**Comment by jasonkarns** [comment]
Looks good so far. Does npm7 document or guarantee that package_json_path will be absolute?

**Comment by rosston** [comment]
@jasonkarns I'm not sure it's a guarantee, but the [RFC](https://github.com/npm/rfcs/blob/latest/implemented/0021-reduce-lifecycle-script-environment.md#implementation) says that `npm_package_json` will come from `@npmcli/run-script`, which [uses `resolve` in its implementation](https://github.com/npm/run-script/blob/3b3281194133db5add16511c81af6ccb31e47484/lib/make-spawn-args.js#L44) and therefore provides an absolute path.

File: lib/load-option.js [hunk 1]
```diff
@@ -3,11 +3,19 @@ var _ = require('lodash')
...
+  if (envVarSet(packageEnvVarName(name))) {
     return boolEnvVarValue(packageEnvVarName(name))
-  } else if (envVarSet(packageArrayEnvVarName(name))) {
+  }
+
+  if (envVarSet(packageArrayEnvVarName(name))) {
     return arrayEnvVarValue(packageEnvVarName(name))
   }
+
+  if (packageJSONPath() && packageJSONScripty()) {
```
Comment (jasonkarns) [inline lib/load-option.js:16]:
```suggestion
  if (packageJsonPath() && packageJsonScripty()) {
```

File: lib/load-option.js [hunk 2]
```diff
@@ -3,11 +3,19 @@ var _ = require('lodash')
...
+    return packageJSONScripty()[name]
```
Comment (jasonkarns) [inline lib/load-option.js:17]:
```suggestion
    return packageJsonScripty()[name]
```

File: lib/load-option.js [hunk 3]
```diff
@@ -49,3 +57,11 @@ function packageEnvVarName (optionName) {
 function packageArrayEnvVarName (optionName) {
   return packageEnvVarName(optionName) + '_0'
 }
+
+function packageJSONPath () {
```
Comment (jasonkarns) [inline lib/load-option.js:61]:
```suggestion
function packageJsonPath () {
```

File: lib/load-option.js [hunk 4]
```diff
@@ -49,3 +57,11 @@ function packageEnvVarName (optionName) {
...
+  return process.env['npm_package_json']
+}
+
+function packageJSONScripty () {
```
Comment (jasonkarns) [inline lib/load-option.js:65]:
```suggestion
function packageJsonScripty () {
```

File: lib/load-option.js [hunk 5]
```diff
@@ -49,3 +57,11 @@ function packageEnvVarName (optionName) {
...
+  return require(packageJSONPath()).scripty
```
Comment (jasonkarns) [inline lib/load-option.js:66]:
```suggestion
  return require(packageJsonPath()).scripty
```

**Review by jasonkarns — COMMENTED** [review]
Just a single nitpick: typically prefer to camelCase function names, which necessitates lowercasing the N+1 chars of an initialism. so `packageJson` instead of `packageJSON`

Do you also have a rationale for expanding the `if`/`elseif` into distinct `if` conditionals? Functionally, they're the same; though "spiritually" they are a single if-elseif statement. If JavaScript had expression-valued-conditionals, ala ruby, it would be a single if-elseif by necessity.

Otherwise, I'm 👍 and appreciative!

**Comment by Chocobozzz** [comment]
> Do you also have a rationale for expanding the if/elseif into distinct if conditionals? Functionally, they're the same; though "spiritually" they are a single if-elseif statement. If JavaScript had expression-valued-conditionals, ala ruby, it would be a single if-elseif by necessity.

Not really, it's just that with 4 ifs I found that it cas more readable to understand the multiple fallbacks. Thanks @jasoncaryallen for the review and the fixes :)

**Comment by Chocobozzz** [comment]
Any news @jasonkarns? :)

**Comment by ben8p** [comment]
Is scripty still maintained ?

This is quite of a blocker for people using npm 7

**Comment by rosston** [comment]
Closing because I think this is resolved in scripty 2.1.0 (via https://github.com/testdouble/scripty/pull/105). Feel free to comment/reopen if this is not actually fixed!

**Pull request closed without merging**